### ISSUE TITLE
Yield test deprecation warning spams too much

### ISSUE THREAD

**Issue by pv** [pytest-dev/pytest#2562]
While working on https://github.com/scipy/scipy/pull/7572 I noticed that pytest emits the "yield tests are deprecated, and scheduled to be removed in pytest 4.0" warning for each generated test, not per each test generator. (And for scipy, this means the message is printed 15652 times :) -- of course we'll eventually plan to refactor them completely away, but that takes some time.)

There are workarounds such as `--disable-warnings` and tagging `| uniq` to the end, but it would be nice if this was fixed to be less spammy on the pytest end. 

As far as I understand, there's no way to disable this deprecation warning in `pytest.ini` without also disabling the whole warning capture mechanism (which is what `--disable-pytest-warnings` et al seem to do)?

Minimal repro:
```
# python -mpytest test_foo.py
def test_foo():
    for k in range(10000):
        yield lambda x: None, k
```
<details>
```
plat: Linux
$ python --version
Python 2.7.13
$ pip list
pip (9.0.1)
py (1.4.34)
pytest (3.1.3)
setuptools (36.0.1)
wheel (0.29.0)
```
</details>

**Comment by nicoddemus** [comment]
Agreed on issuing the warning only once per generator.

To suppress those warnings you can add this to your `pytest.ini`:

```ini
[pytest]
filterwarnings =
    ignore:.*yield tests are deprecated.*
```

**Comment by pv** [comment]
@nicoddemus: adding that to pytest.ini doesn't silence these warnings
for me, in the example case given above.

**Comment by nicoddemus** [comment]
Hmm yeah I didn't test it, sorry. Those parameters are [passed to `warnings.filterwarnings`](https://github.com/pytest-dev/pytest/blob/master/_pytest/warnings.py#L33), so if you can find how to silence them using `warnings.filterwarnings` you should be able to set that in your `pytest.ini`.

**Comment by pv** [comment]
This deprecation warning seems to be emitted at _pytest/python.py:618 directly by `self.config.warn` so it doesn't go through the Python warning mechanism. (Maybe changing it to `warnings.warn` would be enough?)

**Comment by nicoddemus** [comment]
Oh you are right, sorry about that. Indeed we would have to change it to raise an actual warning.

**Comment by nicoddemus** [comment]
xref: #1043

**Comment by pv** [comment]
I'll submit a pull request removing the four spaces in a moment :)
Maybe it makes sense to emit it only once per each generator function.

**Comment by nicoddemus** [comment]
Fixed in #2563
